### PR TITLE
Epmlstrtai 40. The end of Sprint 2.

### PR DESCRIPTION
**Pull request by FunCat**

### CODE REVIEW

File: Java/JDI/jdi-cucumber-test-generator/pom.xml [hunk 1]
```diff
@@ -15,66 +15,105 @@
 
     <dependencies>
         <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.11</version>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/pom.xml]: Extract this into DependencyManagement in parent POM

File: Java/JDI/jdi-cucumber-test-generator/pom.xml [hunk 2]
```diff
@@ -15,66 +15,105 @@
...
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test</artifactId>
+            <version>4.3.10.RELEASE</version>
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/pom.xml]: Extract into Dependency management

File: Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    @BeforeClass
+    public static void setup() {
+        caseController = new CaseController();
+        suitService = new SuitService();
+        suitService.addSuit(new Suit(1, "name1", "first suit description"));
+        suitService.addSuit(new Suit(2, "name2", "second suit description"));
+    }
+
+    @Test
+    public void test1() {
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java]: Give meaningful names to test methods. Here and the next method.

File: Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        caseController.addCaseToSuit(3, case3);
+    }
+
+    @Test
+    public void test2() {
+        Case case1 = new Case((long) 1, "new description", "", suitService.getSuit((long) 1));
+        //caseController.updateCase(case1);
+    }
+
+//    @Test
```
Comment (EvilXa0S) [inline Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java]: Clean up unused code.

File: Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/services/CaseService.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+@Service
+public class CaseService{
+
+    @Autowired
+    private CaseDAO caseDAO;
+
+    @Autowired
+    private SuitDAO suitDAO;
+
+    @Transactional
```
Comment (pomkine) [inline Java/JDI/jdi-cucumber-test-generator/src/main/java/com/epam/test_generator/services/CaseService.java]: You can use this annotation on the class level instead to make all methods transactional

File: Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java (hunk 4 above)
Comment (pomkine) [reply Java/JDI/jdi-cucumber-test-generator/src/test/java/com/epam/test_generator/controllers/CaseControllerTests.java]: Not sure what are we testing here..

**Pull request merged**